### DISCUSSION TITLE
Spot Lock

### DISCUSSION THREAD

**Discussion by murmaider666** [AlexAsplund/Vanchor#11] [Ideas]
Super excited for this project! looking to implement on a Motorguide transom mount pushing a 14' Alumacraft. On my other boat, I have a MinnKota with iPilot on a bow mount. What it doesn't do well all the time is the spot lock feature. It seems to get lost sometimes when there's light gusty wind and starts over correcting. I think its when the stern rotates around the bow and then the bow moves, making the iPilot correct in the wrong direction. What i think would clear this up is an additional GPS sensor on the other end of the boat so the controller always knows which way the boat is pointing. 

Just a thought for future revisions.

**Comment by AlexAsplund** [comment]
Great idea!

I've thought of this as well. But I don't know what's more effective; To have 2 GPS receivers or 1 GPS receiver and a compass.
You could calculate the exact position of your stern/bow (assuming that the GPS offset is constant).

So if you have both your compass and GPS in the center of the bow then your stern is ~14' from current coordinates at heading-180 deg.

But then, you could theoretically substitute a compass with 2 gps receivers I think, given that they have the same accuracy.

(And thank you, now i have the Murmaider song on my brain again)

**Comment by fscognamiglio** [comment]
Hi, Alex. I am trying to know how your interesting code could be evaluated  for a particular use on my boat. I am interested mainly to the hold position function that , I suppose, is activated through the Vanchor service and that should allow steering the electric stepper or a dc motor in order to follow and do mini adjustments in a small circle around the gps  position to hold. Is it right? My issue , moreover, regards how should the engine boat propeller  or rudder be managed for holding a fixed gps position of the boat. 
I thank you very much if you could clarify my ideas and excuse me if my questions are not appropriate. 
Best regards
Francesco

**Comment by fscognamiglio** [comment]
Ok. Excuse me. I believed my question was compatible with the discussion. Inviato dal mio Galaxy
-------- Messaggio originale --------Da: janusz4janusz ***@***.***> Data: 25/01/23  22:57  (GMT+01:00) A: AlexAsplund/Vanchor ***@***.***> Cc: fscognamiglio ***@***.***>, Comment ***@***.***> Oggetto: Re: [AlexAsplund/Vanchor] Spot Lock (Discussion #11) 

I am not Alex. Sorry
